internal/controllers/reconciliation/controller.go

```diff
@@ -304,7 +304,10 @@ func (c *Controller) update(ctx context.Context, resource *resource.Resource, cu
 
 	var patch client.Patch
 	if c.disableSSA {
-		patch = client.MergeFrom(current)
+		// When server-side apply is disabled, use strategic merge patch
+		// This type of patch only adds or updates fields, and never removes fields
+		// that are no longer in the object specification
+		patch = client.StrategicMergeFrom(current)
```

Comment:
Try a different approach. We can't rely on the client's Scheme including the kind of `current`